### ISSUE TITLE
Xiaomi HHCCJCY01HHCC battery

### ISSUE THREAD

**Issue by mihsu81** [theengs/gateway#26]
**Describe the bug**
TheengsGateway doesn't connect to Xiaomi HHCCJCY01HHCC in order to get battery reading like OpenMQTTGateway

**To Reproduce**
1. Start TheengsGateway
2. Have a Xiaomi HHCCJCY01HHCC nearby

**Expected behavior**
TheengsGateway should connect to Xiaomi HHCCJCY01HHCC in order to get battery reading like OpenMQTTGateway.

**Screenshots**

**Environment (please complete the following information):**
* version used (V0.1.5)

**Additional context**

* You should not have a compilation error if you use the versions of the libraries linked into the libraries folder, this badges show you the state of the compilation 
[![Build Status](https://github.com/theengs/decoder/workflows/Build/badge.svg?branch=development)](https://github.com/theengs/decoder/actions)
* If you are not sure this is a bug or an enhancement post your question to the forum

**Comment by 1technophile** [comment]
Indeed, for the moment Theengs focus on ble broadcaster not connection.

**Comment by mihsu81** [comment]
Thanks for the clarification. I thought it's a bug.
TheengsGateway being able to mimic the BLE functionality of OpenMQTTGateway would make it really popular. If you have some pointers about this particular case, I can give it a shot at porting it to TheengsGateway.

**Comment by 1technophile** [comment]
Thanks for proposing help, here is some pointers:
* so as to connect we can already use the model_id coming from the decoder to know if we are dealing with a HHCCJCY01HHCC, 
* once identified OMG attempt a connect for the battery 1/10 times so as to avoid consuming too much battery of the device
* from this you can attempts a connection to it, the gateway uses [bleak](https://bleak.readthedocs.io/en/latest/usage.html), so you can try to use this library
* once the sensor is connected, you will have to retrieve a value from a pair of service/characteristic, [here is](https://github.com/1technophile/OpenMQTTGateway/blob/bf99823861f805ccae4c80b2dc83a5a02b50d2da/main/ZgatewayBLEConnect.ino#L259) where it is done from OMG
* finally this value can be published
This could be done as a separate process from scanning so as to don't block it, even if the data is published with only the battery.

**Comment by mihsu81** [comment]
This is definitely more than I can chew. 😁
Hopefully some of the other developers can add this feature.

**Comment by marcokreeft87** [comment]
I would love this aswell :)

**Comment by marcokreeft87** [comment]
According to the supported devices page this should work, but it doesnt unfortunately
